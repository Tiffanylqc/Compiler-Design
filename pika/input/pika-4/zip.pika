var nn:=[1,5,3,9,2].
var ff:=[2.1,-3.3,4.2,5.0,17.79].
var ratIt:=<int n,float f>->rat{
	return n//[f|int].
}.
var n1:=[1,3,5,7,9,11].
var n2:=[1,2,3,5,7,11].
const makePair:=<int x,int y>->[int]{
	return [x,y].
}.

var a3:= [[^i^,^a^,^m^],[^a^,^m^,^i^]].
var a4:=[[3//4,4//5,5//6],[14//13,14//15,17//16],[14//13,17//16,1//13,24//15,37//16],[1//3,24//5,7//16,24//15,37//16]].
var a2:=[_true_,_true_,_true_,_true_,_false_,_false_].

const f1:=<rat x,rat y>->rat{
	return x*y.
}.

exec{
	print zip nn,ff,ratIt,_n_.
	print zip n1,n2,makePair,_n_.

	var a:=new[[int]](2).
	a:= zip n1,n2,makePair.

	for index i of a4{
		if(i==length a4 - 1){
			break.
		}
		if(length a4[i]==length a4[i + 1]){
			print zip a4[i],a4[i + 1],f1.
		}
	}

	#print zip ff,n2,makePair,_n_.
	#var empty:=new[[int]](3).
	#print zip n2,empty[0],makePair.
}