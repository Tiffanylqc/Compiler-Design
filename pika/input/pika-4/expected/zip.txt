[_1/2, -1_2/3, _3/4, 1_4/5, _2/17]
[[1, 1], [3, 2], [5, 3], [7, 5], [9, 7], [11, 11]]
[_21/26, _56/75, _85/96][_14/39, 5_1/10, _7/208, 2_14/25, 5_89/256]